### PR TITLE
msp430: use assert instead of printf

### PR DESCRIPTION
**Pull request by authmillenon**
Found this while analyzing #3567

### CODE REVIEW

**Comment by OlegHahm** [comment]
Should we postpone this change until next Wednesday?

**Comment by authmillenon** [comment]
What's next Wednesday?

**Comment by kaspar030** [comment]
I'd use assert() instead of the logs. Why waste the flash?

**Comment by OlegHahm** [comment]
Bi-weekly developer meeting: http://riot.pad.spline.de/17

**Comment by authmillenon** [comment]
@kaspar030 at least for the one in L53 it seems not to be an error case, as far as I understand the code.

**Comment by kaspar030** [comment]
@authmillenon Well, it is an unexpected condition that shouldn't happen. If it happens it is a bug. Not what LOG_* should be used for.

That's why assert() would be my choice: if compiled without NDEBUG, silently fix the problem. If compiled by a developer with DEVELHELP, crash so someone gets to know we have a bug...

**Comment by authmillenon** [comment]
Gotcha

**Comment by kaspar030** [comment]
@authmillenon did you try this on an msb430?

**Comment by authmillenon** [comment]
I don't have any MSP-430 board. I just discover those `printf`s that when creating #3567.

**Comment by authmillenon** [comment]
Travis seems to be happy, but this PR is still lacking an ACK.

**Comment by OlegHahm** [comment]
Has someone tested?

File: cpu/msp430-common/lpm_cpu.c [hunk 1]
```diff
@@ -47,16 +48,9 @@ enum lpm_mode lpm_set(enum lpm_mode target)
...
     /* ensure that interrupts are enabled before going to sleep,
        or we're bound to hang our MCU! */
-    if (target != LPM_ON) {
-        if ((__read_status_register() & GIE) == 0) {
-            printf("WARNING: entering low-power mode with interrupts disabled!\n");
-            printf("         Forcing GIE bit to 1!\n\n");
-            __bis_status_register(GIE);
-        }
-    }
+    assert((target == LPM_ON) || (__read_status_register() & GIE));
```
Comment (haukepetersen) [inline cpu/msp430-common/lpm_cpu.c]: This seems to be incorrect? The original statement says 
```c
if ((target != LPM_ON) && (GIE not set)) { -> error; set GIE;}
```
you say
```c
error if ((target != LPM_ON) || (GIE not set))
```
thats a different thing...

Comment (authmillenon) [inline cpu/msp430-common/lpm_cpu.c]: Mine says `if ((target == LPM_ON) || (GIE set)) continue` (note that I don't check if the result of the bit-and is zero. That's logically the same as `if ((target != LPM_ON) && (GIE not set)) abort`. Sure I don't set the GIE bit, see https://github.com/RIOT-OS/RIOT/pull/3569#issuecomment-128086953 why I made this decision.

Comment (haukepetersen) [inline cpu/msp430-common/lpm_cpu.c]: Yep, seems my head is not working correctly today :-) never mind.

**Comment by thomaseichinger** [comment]
Tested manually triggering lpm changes. Seems good.
Go.

**Pull request merged**